var/spack/repos/builtin/packages/petsc/mpi_int64_t_c++11.diff

```diff
@@ -0,0 +1,13 @@
+diff --git a/include/petscsys.h b/include/petscsys.h
+index 933d30bfc8..4c4c991251 100644
+--- a/include/petscsys.h
++++ b/include/petscsys.h
+@@ -294,7 +294,7 @@ M*/
+ .seealso: PetscReal, PetscScalar, PetscComplex, PetscInt, MPIU_REAL, MPIU_SCALAR, MPIU_COMPLEX
+ M*/
+ 
+-#if defined(PETSC_HAVE_STDINT_H) && defined(PETSC_HAVE_INTTYPES_H) && defined(PETSC_HAVE_MPI_INT64_T) /* MPI_INT64_T is not guaranteed to be a macro */
```

Comment:
@pramodk made a good point that we should be careful with this as it might affect other dependencies of/in petsc (e.g. python bindings). I am a bit hesitant whether we should push this diff before it was not approved by PETSc devs

Reply:
This patch is only based on my reading of the headers of mpich 3.2 and HPE-MPI. I want to test against other implementations before contacting PETSc.
 
I think it is more urgent to start evaluating zee performance right now than making sure that the patch works on all super-computers of the world.